File: params/PARAMS_landau_damping.m
n=2^5; 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Map settings
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
params.Nmap         =n; % grid of the submaps
params.Nfine        =2^9; % grid dimension for measuring and hist
params.Nsampling    =2^8; % grid for upsampling the velocity after solving laplace
params.Nplotting    =1024; % plotting grid
params.nv           = n;
params.Lv           = 4*pi;
params.detTol       = 1e-2; % incomp threshold
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% initial condition
params.l            = 1;
params.k            = 0.5;
params.eps          = 5e-2;
params.v0           = 3;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% domain size
params.Lx           = 2*pi/params.k;
params.L = [params.Lx, params.Lv*2];                                        % domain size
params.dom = [0, 0, params.Lx, 2*params.Lv];                                % domain boundaries
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
params.CFL          = 0.8;
params.T_end        = 80;
params.dt           = 0.001/10;
params.iplot        = 100; % plot every iplot time steps
params.ihist        = 100; % only used if dt_hist not set
params.ilog         = 10;  %only used if dt_log not set
params.dt_hist      = 1;
params.dt_log       = 0.1;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
params.bump_transition_width = 0.1*params.Lv;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Options:
params.case            = 'landau_damping';
params.filter          = 'gauss';

File: params/PARAMS_landau_damping.jl
using Parameters

@with_kw struct Params

    n :: Int = 2^5
    Nmap :: Int = n 
    Nfine :: Int = 2^9 
    Nsampling :: Int = 2^8 
    Nplotting :: Int = 1024 
    nv :: Int = n
    Lv :: Float64 = 4 * pi
    detTol :: Float64 = 1e-2 
    l :: Float64 = 1
    k :: Float64 = 0.5
    eps :: Float64 = 5e-2
    v0 :: Float64 = 3
    Lx :: Float64 = 2pi / k
    L :: Vector{Float64} = [Lx, Lv * 2]   
    dom :: Vector{Float64} = [0, 0, Lx, 2 * Lv]    
    CFL :: Float64 = 0.8
    T_end :: Float64 = 80
    dt :: Float64 = 0.001 / 10
    iplot :: Float64 = 100 
    ihist :: Float64 = 100 
    ilog :: Float64 = 10 
    dt_hist :: Float64 = 1
    dt_log :: Float64 = 0.1
    bump_transition_width :: Float64 = 0.1 * Lv
    case :: String = "landau_damping"
    filter :: String = "gauss"

end

params = Params()
